src/python/bot/tasks/train_gradientfuzz_task.py

```diff
@@ -13,9 +13,308 @@
+    logs.log_error('Failed to upload model: cannot get GCS model bucket.')
+    return
+
+  # Zip entire model directory and upload.
+  model_dir_name = os.path.basename(model_directory)
+  zipped_dir = shutil.make_archive(model_dir_name, 'zip', model_directory)
+  gcs_model_path = '{}/{}'.format(gcs_model_directory, zipped_dir)
+
+  logs.log('Uploading the model for fuzzer {} and run {} to {}.'.format(
+      fuzzer_name, model_dir_name, gcs_model_path))
+
+  # Upload files to GCS.
+  result = (storage.copy_file_to(zipped_dir, gcs_model_path))
+
+  if result:
+    logs.log('Uploaded GradientFuzz model {} for fuzzer {}.'.format(
+        model_dir_name, fuzzer_name))
+  else:
+    logs.log_error(
+        'Failed to upload GradientFuzz model {} for fuzzer {}.'.format(
+            model_dir_name, fuzzer_name))
+
+
+def gen_inputs_labels(corpus_directory, fuzzer_binary_path):
+  """
+  Generates inputs and labels from raw input corpus.
+
+  Args:
+    corpus_directory (str): Path to raw inputs.
+    fuzzer_binary_path (str): Path to compiled fuzz target binary.
+
+  Returns:
+    (new_process.ProcessResult): Result of `run_and_wait()`.
+    (str): Dataset name (results stored under
+        GRADIENTFUZZ_SCRIPTS_DIR/data/[dataset_name]).
+  """
+  script_path = get_script_path(run_constants.GENERATE_DATA_SCRIPT)
+  dataset_name = os.path.basename(corpus_directory)
+  args_list = [
+      script_path,
+      run_constants.FUZZ_TARGET_BINARY_FLAG,
+      fuzzer_binary_path,
+      run_constants.INPUT_DIR_FLAG,
+      corpus_directory,
+      run_constants.DATASET_NAME_FLAG,
+      dataset_name,
+      run_constants.MEDIAN_MULT_FLAG,
+      run_constants.DEFAULT_MEDIAN_MULT_CUTOFF,
+  ]
+
+  script_environment = os.environ.copy()
+
+  logs.log('Launching input gen with args: "{}".'.format(str(args_list)))
+
+  # Run process in GradientFuzz directory.
+  data_gen_proc = new_process.ProcessRunner(sys.executable)
+
+  return data_gen_proc.run_and_wait(
+      additional_args=args_list,
+      cwd=GRADIENTFUZZ_SCRIPTS_DIR,
+      env=script_environment,
+      timeout=run_constants.DATA_GEN_TIMEOUT), dataset_name
+
+
+def train_gradientfuzz(fuzzer_name, dataset_name, num_inputs, testing):
+  """Train GradientFuzz model.
+
+  Args:
+    fuzzer_name (str): Prefix to --run-name flag.
+    dataset_name (str): Inputs/labels stored under
+        GRADIENTFUZZ_SCRIPTS_DIR/data/[dataset_name].
+    num_inputs (int): Number of input files (for val split/batch size).
+    testing (bool): Whether or not we use testing number of epochs.
+
+  Returns:
+    (new_process.ProcessResult): Result of `run_and_wait()`.
+    (str): Run name (results stored under
+        GRADIENTFUZZ_SCRIPTS_DIR/models/[architecture]/[run_name]).
+  """
+  if num_inputs < run_constants.MIN_NUM_INPUTS:
+    return new_process.ProcessResult(
+        return_code=run_constants.ExitCode.CORPUS_TOO_SMALL), None
+
+  batch_size = str(4) if testing else str(min(32, int(num_inputs * 0.4)))
```

Comment:
nit: convert these values to a str later (i.e. when given to args_list). That way you only need to str() once and can avoid type confusions in the future since the name implies these to be ints.

Reply:
Gotchu -- makes sense.